app/services/date_formatter.rb

```diff
@@ -0,0 +1,40 @@
+class DateFormatter
+  def initialize(start_date, end_date)
+    @start_date = start_date
+    @end_date = end_date
+  end
+
+  def to_s
+    if start_date == end_date
+      start_date.strftime('%B %d, %Y')
+    else
+      formatted_when
+    end
+  end
+
+  private
+
+  attr_reader :start_date, :end_date
+
+  def formatted_when
+    start_month_text = start_date.strftime('%B')
+    start_date_text = start_date.strftime('%d')
+    start_year_text = start_date.strftime('%Y')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.